core/src/main/java/org/commonjava/indy/core/content/DefaultDownloadManager.java

```diff
@@ -955,6 +962,21 @@ public boolean delete( final ArtifactStore store, final String path, final Event
         return deleted;
     }
 
+    /**
+     * clean just the cache (storage of groups and remote repos)
+     */
+    private boolean deleteCache( ArtifactStore store, String path, EventMetadata eventMetadata )
+                    throws IndyWorkflowException
+    {
+        if ( store.getKey().getType() == hosted )
+        {
+            return false;
+        }
+        final Transfer item = getStorageReference( store, path == null ? ROOT_PATH : path );
```

Comment:
We're caching maven-metadata.xml files in Infinispan now, and I just realized that deleting one of those files will not trigger an update to the Infinispan cache. We need a listener for FileDeletionEvent that will update the @MavenVersionMetadataCache-annotated cache.

Reply:
Thanks. I never thought of it. I will do it.